src/rpc/db/function.c

```diff
@@ -281,7 +281,15 @@ static int db_function_typecheck(string apikey, string name,
         return (-1);
       }
 
-      if (val != args->obj[k].type) {
+      if(val == OBJECT_TYPE_INT && args->obj[k].type == OBJECT_TYPE_UINT){
```

Comment:
Since it's a bit unclear what is happening, I'll summarize it:

+ function with a signed integer argument is registered
+ run request is sent with a positive argument
+ positive argument is unpacked (unpack.c -> unpack_params(..)) into an UNSIGNED integer 

`val` is the argument type we stored in the database. Therefore, since we registered a signed integer, val will be OBJECT_TYPE_INT for this argument. `args->obj[k].data.uinteger > INT64_MAX` checks if the value is a valid *signed integer* packed as *unsigned integer* .